src/odemis/gui/cont/tabs.py

```diff
@@ -4389,7 +4393,474 @@ def _on_align_move_done(self, f=None):
+
+        # Pause all streams
+        self._opt_stream.is_active.value = False
+        self._sem_stream.is_active.value = False
+        self._fib_stream.is_active.value = False
+
+        for view in self.panel.pnl_two_streams_grid.viewports:
+            if view is top['viewport'] or view is bottom['viewport']:
+                view.Shown = True
+            else:
+                view.Shown = False
+        self.view_controller._grid_panel.set_shown_viewports(top['viewport'], bottom['viewport'])
+        self.tab_data_model.visible_views.value = [top['viewport'].view, bottom['viewport'].view]
+
+        # Alter the order of the view_ports so that the correct stream is displayed at top/bottom
+        viewport_list = self.panel.pnl_two_streams_grid.viewports
+        viewport_list.remove(top['viewport'])
+        viewport_list.insert(0, top['viewport'])
+        viewport_list.remove(bottom['viewport'])
+        viewport_list.insert(1, bottom['viewport'])
+
+        self.panel.pnl_two_streams_grid._layout_viewports()
+
+        # Keep a reference to the stream controllers so the garbage collector does not delete them.
+        self._stream_controllers = []
+        # Replace the settings of the top stream controller with the new StreamController
+        for stream_settings in self.panel.top_settings.stream_panels:
+            self.panel.top_settings.remove_stream_panel(stream_settings)
+        new_top_stream_controller = StreamController(self.panel.top_settings,
+                                                     top["stream"],
+                                                     self.tab_data_model,
+                                                     view=top['viewport'].view)
+        new_top_stream_controller.stream_panel.show_remove_btn(False)
+        self._stream_controllers.append(new_top_stream_controller)
+
+        # Replace the settings of the bottom stream controller with the new StreamController
+        for stream_settings in self.panel.bottom_settings.stream_panels:
+            self.panel.bottom_settings.remove_stream_panel(stream_settings)
+        new_bottom_stream_controller = StreamController(self.panel.bottom_settings,
+                                                        bottom["stream"],
+                                                        self.tab_data_model,
+                                                        view=bottom['viewport'].view)
+        new_bottom_stream_controller.stream_panel.show_remove_btn(False)
+        self._stream_controllers.append(new_bottom_stream_controller)
+
+        for v in self.tab_data_model.visible_views.value:
+            if hasattr(v, "stream_classes") and isinstance(top["stream"], v.stream_classes):
+                v.addStream(top["stream"])
+                new_top_stream_controller.stream_panel.show_visible_btn(False)
+            elif hasattr(v, "stream_classes") and isinstance(bottom["stream"], v.stream_classes):
```

Comment:
@pieleric Do I need to check if each view has the attribute stream_classes or can I assume they have and remove "hasattr(view, "stream_classes")" ?